### PR TITLE
pin gfortran to 4.*

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

This PR needs to be merged before https://github.com/conda-forge/staged-recipes/pull/7288.

@isuruf for viz

### CODE REVIEW

**Comment by beckermr** [comment]
@conda-forge-admin, please rerender

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -32,7 +32,7 @@ cxx_compiler:
   - vs2015                     # [win and (environ.get('CF_MAX_PY_VER', '') >= '37')]
 fortran_compiler:              # [unix or win64]
   - toolchain_fort             # [unix and not ppc64le]
-  - gfortran                   # [(linux or osx)                      and (environ.get('CF_COMPILER_STACK') == 'comp7')]
+  - gfortran 4.*               # [(linux or osx)                      and (environ.get('CF_COMPILER_STACK') == 'comp7')]
```
Comment (isuruf) [inline recipe/conda_build_config.yaml]: Shouldn't this be only for osx?

Comment (beckermr) [reply recipe/conda_build_config.yaml]: Ack yes sorry!

Comment (beckermr) [reply recipe/conda_build_config.yaml]: Done! Good catch!

**Comment by isuruf** [comment]
Can you resolve the conflicts?

**Comment by beckermr** [comment]
Ok I think I got this one too!

**Comment by beckermr** [comment]
huzzah!

**Pull request merged**